packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -493,8 +494,16 @@ export abstract class BrowserCrawler<
         const { request, session } = crawlingContext;
 
         if (!request.skipNavigation) {
-            await this._handleNavigation(crawlingContext);
-            tryCancel();
+            try {
+                await this._handleNavigation(crawlingContext);
+                tryCancel();
+            } catch (e: any) {
+                if (this.isProxyError(e)) {
+                    throw new SessionError();
+                } else {
+                    throw e;
+                }
+            }
```

Comment:
can we move this into the `_handleNavigation` method? i guess this can come only from the `_navigationHandler` call we already wrap with try/catch?